src/plugins/dashboard/public/dashboard_actions/expand_panel_action.tsx

```diff
@@ -51,7 +51,8 @@ export class ExpandPanelAction implements Action<EmbeddableApiContext> {
   public async execute({ embeddable }: EmbeddableApiContext) {
     if (!isApiCompatible(embeddable)) throw new IncompatibleActionError();
     embeddable.parentApi.expandPanel(
-      embeddable.parentApi.expandedPanelId.value ? undefined : embeddable.uuid
+      embeddable.parentApi.expandedPanelId.value ? undefined : embeddable.uuid,
+      embeddable.parentApi.expandedPanelId.value
```

Comment:
I need the `previousPanelId` here to pass into `setScrollPanelId` which needs to be set because  the `dashboard_grid_item` useEffect only calls `scrollToPanel` when `scrollPanelId` is defined (see https://github.com/elastic/kibana/pull/184696/files#diff-7c3ce7a8fd548fbdb29c73c2dd97d6a25f952c094bf06e9572cd2084a61ed323R68-R70). This may not be the most elegant way to handle this.

Reply:
In `expandPanel` you could just do something like this then, no?

```typescript
  public expandPanel = (panelId?: string, expanded?: boolean) => {
    this.setExpandedPanelId(expanded ? panelId : undefined);

    if (panelId && window.scrollY > 0) {
      this.scrollPosition = window.scrollY;
      return;
    }

    if (expanded) {
      this.setScrollToPanelId(panelId);
    }
  };
```

Reply:
Actually you're right. We could pass the `panelId` always, and we could even skip passing the boolean and grab the current value of `expandedPanelId` inside of `expandPanel` to determine if we're expanding or collapsing. 

So like this:

```typescript
  public expandPanel = (panelId: string) => {
    const isPanelExpanded = Boolean(this.getExpandedPanelId());

    if (isPanelExpanded) {
      this.setExpandedPanelId(undefined);
      this.setScrollToPanelId(panelId);
      return;
    }

    this.setExpandedPanelId(panelId);
    if (window.scrollY > 0) {
      this.scrollPosition = window.scrollY;
    }
  };
```